crates/rustc_codegen_spirv/src/spirv_type.rs

```diff
@@ -276,6 +276,42 @@ impl SpirvType<'_> {
         id
     }
 
+    /// Returns how many Input / Output `location`s this type occupies, or None if this type is not allowed to be sent.
+    ///
+    /// See [Vulkan Spec 16.1.4. Location and Component Assignment](https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#interfaces-iointerfaces-locations)
+    #[allow(clippy::match_same_arms)]
+    pub fn location_size(&self, cx: &CodegenCx<'_>) -> Option<u32> {
+        let result = match *self {
+            // bools cannot be in an Input / Output interface
+            Self::Bool => return None,
+            Self::Integer(_, _) | Self::Float(_) => 1,
+            Self::Vector { .. } => 1,
+            Self::Adt { field_types, .. } => {
+                let mut locations = 0;
+                for f in field_types {
+                    locations += cx.lookup_type(*f).location_size(cx)?;
+                }
+                locations
+            }
+            Self::Matrix { element, count } => cx
+                .lookup_type(element)
+                .location_size(cx)?
+                .checked_mul(count)
+                .expect("overflow"),
```

Comment:
This is not really more helpful than `.unwrap()`. Neither in runtime not during code review.

The best approach so far I have found in blockchain space:
* write the code in a way that expects no panics at all
* only use `.expect()` where panics never happen and in its message you write a proof that reviewer can verify

For example:
```rust
let n = NonZeroUsize::new(1).expect("Not zero; qed");
```